x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -722,7 +722,7 @@ protected LogicalPlan rule(LogicalPlan plan) {
             }
 
             // Try to resolve aggregates and groupings based on the child plan
-            if (plan instanceof Aggregate) {
+            if (plan instanceof Aggregate && !plan.resolved() && plan.childrenResolved()) {
```

Comment:
```suggestion
            if (plan instanceof Aggregate && plan.resolved() == false && plan.childrenResolved()) {
```
despite my own prefs.

Reply:
Considering this condition applies for all ifs inside this rule, I would extract it to the top:
```java
        @Override
        protected LogicalPlan rule(LogicalPlan plan) {
            if (plan.resolved() || plan.childrenResolved() == false) {
               return plan;
            }
            if (plan instanceof Filter) {...

```

In fact the check for children resolution is present I believe in all analyzer rules so it might make to create a subclass that all other rules extend to avoid having to repeat this bit of code which as we can see, introduces some subtle bugs.

I'm fine with doing it in a follow-up PR to get this in (with the if moved to the top level).